packages/fast-usdc/src/exos/status-manager.js

```diff
@@ -1,12 +1,205 @@
+       *
+       * @param {NobleAddress} address
+       * @param {bigint} amount
+       * @param {number} index
+       */
+      advance(address, amount, index) {
+        const key = toStatusKey(address, amount);
+        const mutableEntries = [...txs.get(key)];
+        if (!mutableEntries[index]) {
+          throw makeError(`Cannot find ${q(key)} entry at index ${q(index)}`);
+        }
+        if (mutableEntries[index].status !== TxStatus.Observed) {
+          throw makeError(
+            `Cannot advance ${q(key)} entry ${q(index)} with status: ${q(mutableEntries[index].status)}`,
+          );
+        }
+        mutableEntries[index] = {
+          ...mutableEntries[index],
+          status: TxStatus.Advanced,
+        };
+        txs.set(key, harden(mutableEntries));
+      },
+      /**
+       * Mark an `OBSERVED` transaction as `SKIPPED`.
+       *
+       * @param {NobleAddress} address
+       * @param {bigint} amount
+       * @param {number} index
+       */
+      skip(address, amount, index) {
+        const key = toStatusKey(address, amount);
+        const mutableEntries = [...txs.get(key)];
+        if (!mutableEntries[index]) {
+          throw makeError(`Cannot find ${q(key)} entry at index ${q(index)}`);
+        }
+        // TODO, consider ADVANCED -> SKIPPED for failed IBC Transfer
+        if (mutableEntries[index].status !== TxStatus.Observed) {
+          throw makeError(
+            `Cannot skip ${q(key)} entry ${q(index)} with status: ${q(mutableEntries[index].status)}`,
+          );
+        }
+        mutableEntries[index] = {
+          ...mutableEntries[index],
+          status: TxStatus.Skipped,
+        };
+        txs.set(key, harden(mutableEntries));
+      },
+      /**
+       * Find an `ADVANCED` or `SKIPPED` tx waiting to be `SETTLED`
+       *
+       * @param {NobleAddress} address
+       * @param {bigint} amount
+       * @returns {boolean}
+       */
+      hasPendingSettlement(address, amount) {
+        const key = toStatusKey(address, amount);
+        const entries = txs.get(key);
+        const unsettledIdx = getPendingIdx(entries);
+        return unsettledIdx > -1;
+      },
+      /**
+       * Mark an `ADVANCED` transaction as `SETTLED`.
+       *
+       * TODO - does this need to support `OBSERVED` -> `SETTLED`?
+       *
+       * @param {NobleAddress} address
+       * @param {bigint} amount
+       */
+      settle(address, amount) {
+        const key = toStatusKey(address, amount);
+        const mutableEntries = [...txs.get(key)];
+        const unsettledIdx = getPendingIdx(mutableEntries);
+        if (unsettledIdx === -1) {
+          throw makeError(`No unsettled entry for ${q(key)}`);
+        }
+        // TODO, vstorage update for `TxStatus.Settled`
+        // delete entry once SETTLED
+        mutableEntries.splice(unsettledIdx, 1);
+        txs.set(key, harden(mutableEntries));
+      },
+      /**
+       * @overload
```

Comment:
TIL! you can also do this in a TS expression inside a `@type` comment. give that a try because it would be harder to miss. (My eyes only looked up to the lowest `/**` and didn't think to keep searching)

Reply:
No luck with tihs:

```js
/**
 * @type {((address: NobleAddress, amount: bigint) => StatusManagerValue[]) | ((address: NobleAddress, amount: bigint, index: number) => StatusManagerValue | undefined)}
 */
```

Was also playing around with these:

```js
/**
 * @type {<T extends number | undefined = undefined>(
 *   address: NobleAddress, 
 *   amount: bigint, 
 *   index?: T
 * ) => T extends number ? StatusManagerValue | undefined : StatusManagerValue[]}
 */
```

```js
/**
 * @template {number | undefined} [T=undefined]
 * @param {NobleAddress} address
 * @param {bigint} amount
 * @param {T} [index]
 * @returns {T extends number ? StatusManagerValue | undefined : StatusManagerValue[]}
 */
 ```

Reply:
I also don't know if the `index` logic in `StatusManager` is here to say. Some of the convos in other threads lead me to believe we won't need it. So not going to spin wheels too hard on this rn.